frontend/packages/schema-editor/src/components/layout/TypesPanel.tsx

```diff
@@ -1,36 +1,28 @@
 import React from 'react';
-import { useDispatch } from 'react-redux';
-import {
-  setSelectedAndFocusedNode,
-  setSelectedNode,
-} from '@altinn/schema-editor/features/editor/schemaEditorSlice';
 
 import type { UiSchemaNode } from '@altinn/schema-model';
-import { FieldType, ObjectKind, isCombination } from '@altinn/schema-model';
+import { FieldType, ObjectKind } from '@altinn/schema-model';
 
 import { useTranslation } from 'react-i18next';
 import { ActionMenu } from '../common/ActionMenu';
 import { IconImage } from '../common/Icon';
 import { useAddProperty } from '../../hooks/useAddProperty';
 import { SchemaTree } from '../SchemaTree';
+import { useSchemaEditorAppContext } from '@altinn/schema-editor/hooks/useSchemaEditorAppContext';
 
 export type TypesPanelProps = {
   uiSchemaNode: UiSchemaNode;
 };
 export const TypesPanel = ({ uiSchemaNode }: TypesPanelProps) => {
   const translation = useTranslation();
   const t = (key: string) => translation.t('schema_editor.' + key);
-  const dispatch = useDispatch();
+  const { setSelectedNodePointer } = useSchemaEditorAppContext();
   const addProperty = useAddProperty();
 
   const handleAddProperty = (objectKind: ObjectKind, fieldType?: FieldType) => {
     const newPointer = addProperty(objectKind, fieldType, uiSchemaNode.pointer);
     if (newPointer) {
-      dispatch(
-        isCombination(uiSchemaNode)
-          ? setSelectedNode(newPointer)
-          : setSelectedAndFocusedNode(newPointer),
-      );
+      setSelectedNodePointer(newPointer);
```

Comment:
Is it possible to test this? 😄

Reply:
Also out of scope, and the component will be rewritten in https://github.com/Altinn/altinn-studio/issues/11910.